src/r-bridge/lang-4.x/ast/parser/json/format.ts

```diff
@@ -51,3 +76,47 @@ export function prepareParsedData(data: string): Entry[] {
 
 	return roots
 }
+
+export function convertPreparedParsedData(roots: CsvEntry[]): JsonEntry {
+	// Locate start, end of source file (order children in advance).
+	const rootEntries = roots.sort(orderOf)
+	const start = rootEntries[0]
+	const end = rootEntries[rootEntries.length - 1]
+
+	// Construct CsvEntry for the root, handling empty input.
+	const csvParent: CsvEntry = {
```

Comment:
i still think we should have a separate check for the length of the children. if we have none (i.e., there is no start...) we can cheap out of the recursive traversal and no loner require 4 / 2 null checks.